presto-main/src/main/java/io/prestosql/sql/planner/QueryPlanner.java

```diff
@@ -751,112 +553,94 @@ private PlanBuilder window(PlanBuilder subPlan, List<FunctionCall> windowFunctio
             return subPlan;
         }
 
-        for (FunctionCall windowFunction : windowFunctions) {
+        for (FunctionCall windowFunction : scopeAwareDistinct(subPlan, windowFunctions)) {
             Window window = windowFunction.getWindow().get();
```

Comment:
Filters are not legal for window functions. I'll add a check to verify they are not present.

Reply:
I tried to keep this code as untouched as possible, but I agree it can be cleaned up a bit. I'll change how the function is planned -- in particular, relying on the rewrite and then extract the pieces that matter is not ideal. In the future (once we split AST from IR), there may not be a "FunctionCall" representation for window functions.